### PR TITLE
feat(core): support for nested nx projects

### PR DESCRIPTION
**Pull request by AgentEnder**
## Current Behavior
Some utility methods don't properly exclude files that are part of a nested project from the container project

## Expected Behavior
Utility methods exclude files that are from a child project from the parent.

## Related Issue(s)

Fixes #

### CODE REVIEW

File: packages/nx/src/config/workspaces.ts [hunk 1]
```diff
@@ -603,24 +603,50 @@ export function getGlobPatternsFromPackageManagerWorkspaces(
...
+      }
     }
-  } catch {
-    return undefined;
-  }
+
+    if (existsSync(join(root, 'lerna.json'))) {
+      try {
+        const { packages } = readJsonFile<any>(join(root, 'lerna.json'));
+        patterns.push(packages?.length > 0 ? normalizePatterns(packages) : []);
```
Comment (vsavkin) [inline packages/nx/src/config/workspaces.ts:632]: IT has to be packages/*. I believe hta'ts the default lerna uses

**Review by vsavkin — CHANGES_REQUESTED** [review]

File: packages/nx/src/generators/utils/project-configuration.ts [hunk 2]
```diff
@@ -441,7 +444,8 @@ function findCreatedProjects(tree: Tree) {
       const fileName = basename(f.path);
       return (
         f.type === 'CREATE' &&
-        (fileName === 'project.json' || fileName === 'package.json')
+        (fileName === 'project.json' || fileName === 'package.json') &&
+        f.path !== 'package.json'
```
Comment (FrozenPandaz) [inline packages/nx/src/generators/utils/project-configuration.ts:448]: This should be excluded only if the env variable is OFF

File: packages/nx/src/generators/utils/project-configuration.ts [hunk 3]
```diff
@@ -433,6 +433,9 @@ function inlineProjectConfigurationsWithTree(
  * the same devkit generator run show up when
  * there is no workspace.json file, as `glob`
  * cannot find them.
+ *
+ * We exclude the root `package.json` from this list, since it wouldn't be
```
Comment (FrozenPandaz) [inline packages/nx/src/generators/utils/project-configuration.ts:437]: This comment should stay with the exclusion line

**Review by FrozenPandaz — APPROVED** [review]

File: packages/nx/src/config/workspaces.ts (hunk 1 above)
Comment (AgentEnder) [reply packages/nx/src/config/workspaces.ts:632]: Default wasn't mentioned it lerna's docs, I'll add it here, moving it from below.

File: packages/nx/src/project-graph/affected/locators/workspace-projects.ts [hunk 4]
```diff
@@ -15,15 +15,22 @@ export const getTouchedProjects: TouchedProjectLocator = (
...
-  return touchedFiles
-    .map((f) => {
-      return projectNames.find((projectName) => {
-        const p = projectGraphNodes[projectName].data;
-        const projectRoot = p.root.endsWith('/') ? p.root : `${p.root}/`;
-        return f.file.startsWith(projectRoot);
-      });
-    })
-    .filter(Boolean);
+  return touchedFiles.reduce((affected, f) => {
```
[the remark this answers is not in the document]
Comment (AgentEnder) [reply packages/nx/src/project-graph/affected/locators/workspace-projects.ts:18]: Resolved by reusing utils from project locator

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**